### PR TITLE
Fix a bug in gourp_points_by_bounding_boxes data augmentation

### PR DESCRIPTION
**Pull request by lengzq**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by lengzq** [comment]
@tanzhenyu

File: keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes.py [hunk 1]
```diff
@@ -231,16 +231,13 @@ def call(self, inputs, training=True):
...
-                    object_point_clouds_list += [object_point_clouds[tf.newaxis, ...]]
-                    object_bounding_boxes_list += [
-                        object_bounding_boxes[tf.newaxis, ...]
-                    ]
-
+                    object_point_clouds_list += [object_point_clouds]
+                    object_bounding_boxes_list += [object_bounding_boxes]
                 inputs[OBJECT_POINT_CLOUDS] = tf.concat(
-                    object_point_clouds_list, axis=0
+                    object_point_clouds_list, axis=1
```
Comment (tanzhenyu) [inline keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes.py:237]: It'd be slightly less error-prone to use `axis=-2`?

Comment (lengzq) [reply keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes.py]: Done.

File: keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes.py [hunk 2]
```diff
@@ -80,6 +80,13 @@ def __init__(
         self._max_points_per_bounding_boxes = max_points_per_bounding_boxes
         self._auto_vectorize = False
 
+    def get_params(self):
+        return "label_index:{},min_points_per_bounding_boxes:{},max_points_per_bounding_boxes:{}".format(
+            str(self._label_index),
+            str(self._min_points_per_bounding_boxes),
```
Comment (tanzhenyu) [inline keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes.py:86]: Actually -- you should probably override `get_config` for layers:
https://keras.io/guides/serialization_and_saving/

File: keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes.py [hunk 3]
```diff
@@ -231,16 +238,13 @@ def call(self, inputs, training=True):
...
-                    object_point_clouds_list += [object_point_clouds[tf.newaxis, ...]]
-                    object_bounding_boxes_list += [
-                        object_bounding_boxes[tf.newaxis, ...]
-                    ]
-
+                    object_point_clouds_list += [object_point_clouds]
+                    object_bounding_boxes_list += [object_bounding_boxes]
                 inputs[OBJECT_POINT_CLOUDS] = tf.concat(
-                    object_point_clouds_list, axis=0
+                    object_point_clouds_list, axis=-3
```
Comment (tanzhenyu) [inline keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes.py:244]: maybe comment on what -3 means here?
same for -2 below

File: keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes_test.py [hunk 4]
```diff
@@ -26,6 +26,17 @@
 
 
 class GroupPointsByBoundingBoxesTest(tf.test.TestCase):
+    def test_get_params(self):
```
Comment (tanzhenyu) [inline keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes_test.py:29]: one more thing -- can you update the training script to global_batch > 1, and verify it works?

**Review by tanzhenyu — CHANGES_REQUESTED** [review]

File: keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes.py (hunk 2 above)
Comment (lengzq) [reply keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes.py]: Done.

File: keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes_test.py (hunk 4 above)
Comment (tanzhenyu) [reply keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes_test.py]: I will do this in next PR then

File: keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes_test.py [hunk 5]
```diff
@@ -26,6 +26,21 @@
 
 
 class GroupPointsByBoundingBoxesTest(tf.test.TestCase):
+    def test_get_config(self):
+        add_layer = GroupPointsByBoundingBoxes(
+            label_index=1,
+            min_points_per_bounding_boxes=1,
+            max_points_per_bounding_boxes=4,
+        )
```
[the remark this answers is not in the document]
Comment (lengzq) [reply keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes_test.py:34]: Done.

File: keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes.py (hunk 3 above)
Comment (lengzq) [reply keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes.py:244]: Added a comment.

**Review by tanzhenyu — APPROVED** [review]
